codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -132,8 +140,7 @@ public static ByteBuf encode(
         if (e > 1 && dest.getByte(e - 1) == NEW_LINE) {
             e--;
         }
-
-        return dest.slice(0, e);
+        return destBuf.slice(0, e);
```

Comment:
We should still call dst.release() if it's not the same as destBuf

Reply:
And...if `e` isn't changed, there's no need to have a slice either (although not a biggie)

Reply:
The writer index isn't updated by the encoding, which is what the slice was doing. We can just call `setIndex`.